Library/Homebrew/extend/os/mac/formula.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Formula
+  class << self
+    def uses_from_macos(dep, **args)
+      depends_on(dep) if add_mac_dependency?(args)
+    end
+
+    private
+
+    def add_mac_dependency?(args)
+      args.each { |key, version| args[key] = OS::Mac::Version.from_symbol(version) }
+
+      return false if args[:after] && OS::Mac.version < args[:after]
+
+      return false if args[:before] && OS::Mac.version >= args[:before]
+
+      true
```

Comment:
Perhaps…
```ruby
  !(args[:after] && OS::Mac.version < args[:after] ||
    args[:before] && OS::Mac.version >= args[:before])
```
I'm fine with either. Which do you prefer?

Reply:
Maybe we should stick with the early return solution, it's more objective than doing the previous evaluation

Reply:
I prefer the current version 👍